File: 728-self-dividing-numbers/728-self-dividing-numbers.cpp
class Solution {
    public:
    bool check(int i){
        if(i==0)
            return false;
        else{
        int temp = i;
        while(temp!=0){
           int rem = temp%10;
            if(rem==0)
                return false;
            if(i%rem!=0)
                return false;
            temp = temp/10;
        }
        return true;
        }
    }
public:
    vector<int> selfDividingNumbers(int left, int right) {
        vector<int> ans;
         for(int i=left;i<=right;i++){
             if(check(i))
                 ans.push_back(i);  
         }
        return ans;
    }
};

File: 728-self-dividing-numbers/728-self-dividing-numbers.java
class Solution {
    public List<Integer> selfDividingNumbers(int left, int right) {
        ArrayList<Integer> arr = new ArrayList<>();
        for(int i=left;i<=right;i++){
            if(solve(i)){
                arr.add(i);
            }
        }
        return arr;
    }
    
    static boolean solve(int n){
        boolean flag = true;
        int temp = n;
        while(n>0){
            int rem = n%10;
            if(rem==0){
                return false;
            }
            if(temp%rem != 0){
               return false;
            }
            n = n/10;
        }
         return flag;
    }
}